### PR TITLE
Further specify inclusive/exclusive datetime values for BillingPeriod{Start,End} and ChargePeriod{Start,End}

### PR DESCRIPTION
**Pull request by cnharris10**
Issue: https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/380

# Change

Further clarify within the column definition that various columns MUST contain inclusive/exclusive datetimes:
- BillingPeriodStart -> inclusive
- BillingPeriodEnd -> exclusive
- ChargePeriodStart -> inclusive
- ChargePeriodEnd -> exclusive

### CODE REVIEW

**Comment by jpradocueva** [comment]
The group suggested adding an example and a clear definition. 
The terms are ok.

**Comment by mike-finopsorg** [comment]
Discussion in weekly meeting was we need more clarity on what 'inclusive' and 'exclusive' means. Either inline on the column definition or by adding entries in the Glossary `Inclusive Date` and `Exclusive Date` and linking. It would help to provide an example like
```
Inclusive.      Exclusive
1/03/2024 - 1/04/2024

Will include the 1/03 but exclude the 1/04
```

**Comment by cnharris10** [comment]
Will add an example to this pull request.

File: specification/glossary.md [hunk 1]
```diff
@@ -60,6 +60,10 @@ A specification-defined categorical attribute that provides context or categoriz
 
 Cost inclusive of the impacts of all reduced rates and discounts, augmented with the amortization of relevant purchases (one-time or recurring) paid to cover future eligible charges.
 
+<a name="glossary:exclusivebound"><b>Exclusive Bound</b></a>
+
+A Date/Time Format value that is not contained within the ending bound of a time period.
+
```
Comment (udam-f2) [inline specification/glossary.md:66]:
```suggestion
<a name="glossary:exclusivebound"><b>Exclusive Date/Time Value</b></a>

A date/time value that is exclusive can be used to represent the end of a period where the period ends immediately before the specified date/time. An exclusive date/time is in contrast to an [inclusive date/time](#glossary:inclusivebound), which is commonly used to signify the start of a period.

```
Suggestion. Ignore if it doesn't improve. If accepted, the links to this glossary will need to be updated.

File: specification/glossary.md [hunk 2]
```diff
@@ -68,6 +72,10 @@ A tag with one tag value chosen from a set of possible tag values after being pr
 
 An open-source specification that defines requirements for billing data.
 
+<a name="glossary:inclusivebound"><b>Inclusive Bound</b></a>
+
+A Date/Time Format value that is contained within the beginning bound of a time period.
```
Comment (udam-f2) [inline specification/glossary.md:77]:
```suggestion
<a name="glossary:inclusivebound"><b>Inclusive Date/Time Value</b></a>

A date/time value that is inclusive can be used to represent the start of a period where the period starts at the specified date/time. An inclusive date/time is in contrast to an [exclusive date/time](#glossary:exclusivebound) which is commonly used to signify the end of a period.
```

File: specification/columns/billingperiodend.md [hunk 3]
```diff
@@ -2,7 +2,11 @@
...
 
-The BillingPeriodEnd column MUST be present in the billing data. This column MUST be of type Date/Time and MUST NOT contain null values. BillingPeriodEnd column MUST conform to [Date/Time Format](#date/timeformat) requirements. The sum of the [BilledCost](#billedcost) column for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
+The BillingPeriodEnd column MUST be present in the billing data. This column MUST be of type Date/Time and MUST NOT contain null values.
+
+BillingPeriodEnd column MUST conform to an [Exclusive](#glossary:exclusivebound) [Date/Time Format](#date/timeformat) requirements. The sum of the [BilledCost](#billedcost) column for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
+
+For example, a time period where BillingPeriodStart is '2024-01-01T00:00:00Z' and BillingPeriodEnd is '2024-02-01T00:00:00Z' only includes charges for January, given BillingPeriodEnd is *excluded* from the period.
```
Comment (udam-f2) [inline specification/columns/billingperiodend.md:9]: Should the example be a part of the description paragraph so we can leave the normative content as one paragraph at the end? Just styling-wise, this seems a bit different from our standard approach

File: specification/columns/billingperiodend.md [hunk 4]
```diff
@@ -14,7 +18,7 @@ Billing Period End
 
 ## Description
 
-The end date and time of the *billing period*.
+The exclusive, end date and time of the *billing period*.
```
Comment (udam-f2) [inline specification/columns/billingperiodend.md:21]: Not an expert but I believe the comma here is grammatically incorrect.

File: specification/columns/billingperiodend.md [hunk 5]
```diff
@@ -1,8 +1,10 @@
...
 
-The BillingPeriodEnd column MUST be present in the billing data. This column MUST be of type Date/Time and MUST NOT contain null values. BillingPeriodEnd column MUST conform to [Date/Time Format](#date/timeformat) requirements. The sum of the [BilledCost](#billedcost) column for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
+The BillingPeriodEnd column MUST be present in the billing data. This column MUST be of type [Date/Time Format](#date/timeformat), MUST be an *exclusive* value, and MUST NOT contain null values.
+
+The sum of the [BilledCost](#billedcost) column for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
```
Comment (udam-f2) [inline specification/columns/billingperiodend.md:7]: Can we either keep the normative stuff in one paragraph or as bullets?
```suggestion
The BillingPeriodEnd column MUST be present in the billing data. This column MUST be of type [Date/Time Format](#date/timeformat), MUST be an *exclusive* value, and MUST NOT contain null values. The sum of the [BilledCost](#billedcost) column for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
```

**Review by ahullah — APPROVED** [review]

File: specification/columns/billingperiodstart.md [hunk 6]
```diff
@@ -1,8 +1,10 @@
...
 
-The BillingPeriodStart column MUST be present in the billing data. This column MUST be of type Date/Time and MUST NOT contain null values. BillingPeriodStart column MUST conform to [Date/Time Format](#date/timeformat) requirements. The sum of the [BilledCost](#billedcost) metric for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
+The BillingPeriodStart column MUST be present in the billing data, MUST be of type [Date/Time Format](#date/timeformat), MUST be an *inclusive* value, and MUST NOT contain null values.
+
+The sum of the [BilledCost](#billedcost) metric for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
```
Comment (udam-f2) [inline specification/columns/billingperiodstart.md:7]:
```suggestion
The BillingPeriodStart column MUST be present in the billing data, MUST be of type [Date/Time Format](#date/timeformat), MUST be an *inclusive* value, and MUST NOT contain null values. The sum of the [BilledCost](#billedcost) metric for [*rows*](#glossary:row) in a given *billing period* MUST match the sum of the invoices received for that *billing period* for a [*billing account*](#glossary:billing-account).
```

**Review by udam-f2 — APPROVED** [review]

**Comment by jpradocueva** [comment]
Approved by the group on April 25.

**Comment by jpradocueva** [comment]
@cnharris10 Please address the final comments and merge the PR. Thanks.

File: specification/columns/billingperiodend.md (hunk 4 above)
Comment (jpradocueva) [inline specification/columns/billingperiodend.md]:
```suggestion
The exclusive end date and time of the *billing period*.
```

**Review by jpradocueva — APPROVED** [review]
Approved the change to `The exclusive end date and time of the *billing period*`

**Review by jpradocueva — APPROVED** [review]
Two conversations were resolved.

**Pull request merged**